src/librbd/api/Mirror.cc

```diff
@@ -651,7 +651,13 @@ void Mirror<I>::image_get_info(I *ictx, mirror_image_info_t *mirror_image_info,
   auto req = mirror::GetInfoRequest<I>::create(*ictx, &ctx->mirror_image,
                                                &ctx->promotion_state,
                                                ctx);
-  req->send();
+  if (ictx->state->is_refresh_required()) {
```

Comment:
Nit: just use `ictx->state->refresh_if_required()`

Reply:
We don't have async version of `refresh_if_required`. Do you want me to add it?

Reply:
Sorry, didn't realize that this was an async API method.  In reality, if the refresh fails we should not advance. That would then imply that we need to delay the creation of `C_ImageGetInfo` until after the refresh succeeds (if a refresh is required).

Reply:
Just to make sure I understand what you suggest. Should I implement the refresh retry loop until it succeeds? Wouldn't just checking for refresh error and returning it be enough? Why this image_get_info so special? I don't see we do something like this in other places.

And what if the error is fatal, e.g. due to the client is blacklisted or the image is removed? We may loop forever.

Reply:
Right now, sync API methods call `refresh_if_required` and return the error code if the refresh failed. Therefore, if `is_refresh_required` is `true` and then `refresh` fails, you should return that failure code to the async caller.